chainer/functions/array/spatial_transformer_grid.py

```diff
@@ -62,13 +63,13 @@ def _forward(self, inputs):
         xp = cuda.get_array_module(theta)
 
         ys, xs = xp.meshgrid(
-            xp.linspace(-1, 1, H, dtype=numpy.float32),
-            xp.linspace(-1, 1, W, dtype=numpy.float32), indexing='ij',
+            xp.linspace(-1, 1, H, dtype=self._dtype),
+            xp.linspace(-1, 1, W, dtype=self._dtype), indexing='ij',
```

Comment:
Use `theta.dtype` instead of `self._dtype`? (ditto for other parts, too.) We could remove `self._dtype`, too.